File: app/src/main/java/com/xia/banner/option/text/NoticeHolder.java
package com.xia.banner.option.text;

import android.view.View;

import com.xia.banner.R;
import com.xia.flybanner.holder.FBHolder;

import androidx.appcompat.widget.AppCompatTextView;

/**
 * @author weixia
 * @date 2019/4/16.
 */
public class NoticeHolder extends FBHolder {
    private AppCompatTextView mAppCompatTextView;

    NoticeHolder(View itemView) {
        super(itemView);
    }

    @Override
    protected void initView(View itemView) {
        mAppCompatTextView = itemView.findViewById(R.id.item_notice_tv);
    }

    @Override
    public void updateUI(Object data) {
        if (mAppCompatTextView != null) {
            mAppCompatTextView.setText(String.valueOf(data));
        }
    }
}

File: app/src/main/java/com/xia/banner/option/text/NoticeHolder.kt
package com.xia.banner.option.text

import android.view.View

import com.xia.banner.R
import com.xia.flybanner.holder.FBHolder

import androidx.appcompat.widget.AppCompatTextView

/**
 * @author weixia
 * @date 2019/4/16.
 */
class NoticeHolder internal constructor(itemView: View) : FBHolder<Any>(itemView) {
    private var textView: AppCompatTextView? = null

    override fun initView(itemView: View) {
        textView = itemView.findViewById(R.id.item_notice_tv)
    }

    override fun updateUI(data: Any) {
        textView?.text = data.toString()
    }
}
